### ISSUE TITLE
Search in Russian: consider "е" and "ё" interchangable in Russian

### ISSUE THREAD

**Issue by aleksejrs** [openfoodfacts/openfoodfacts-server#455]
Find "е" when searching for "ё", and "ё" when searching for "е".

**Comment by stephanegigandet** [comment]
Are those the only interchangeable characters?

And can you think of any conflicts where a word/name with "e" means something different than the same word with "ë"?

**Comment by aleksejrs** [comment]
> Are those the only interchangeable characters?

Yes, for letters.  "ё"/"Ё" has usually been written as "е"/"Е". https://en.wikipedia.org/wiki/Yo_(Cyrillic)#Russian
As for other characters, there is, of course, punctuation like "«»", "„“" etc.

> And can you think of any conflicts where a word/name with "e" means something different than the same word with "ë"?

That is a problem, but probably not one we can do anything about. It will probably mostly affect place names.

**Comment by teolemon** [comment]
fr:pâtés
fr:pates

**Comment by aleksejrs** [comment]
I am talking only about search (where you enter search keywords and get results, not even graphs).

There has already been a problem which resulted in some labels or categories to have "и" instead of "й".

**Comment by aleksejrs** [comment]
Although categories might have no problem if е/ё synonyms are auto-generated.  I don't know.

**Comment by hangy** [comment]
We need to see if this is something that needs to be implemented locale-aware. We have had bug reports where "de:Bürger" was automatically deaccentuated to "de:Burger", even though those are not interchangeable.

**Comment by aleksejrs** [comment]
"Ё" is a separate letter, not a "е" with an umlaut.

**Comment by hangy** [comment]
True, in German they are just vowels with umlauts, but they canned be used interchangeably either. In Finnish or Swedish, "Ö" is considered a separate letter, too, but deaccentuated by OFF all the same. Not saying we should not do it, I just want to raise caution. 👍

**Comment by aleksejrs** [comment]
You could instead enforce the use of "ё" where the label says "е" instead of "ё". ;)
